src/Nethermind/Nethermind.Network/PeerPool.cs

```diff
@@ -185,23 +186,28 @@ private void StartPeerPersistenceTimer()
                 try
                 {
                     _peerPersistenceTimer.Enabled = false;
-                    RunPeerCommit();
+                    _storageCommitTask = RunPeerCommit().ContinueWith(x =>
+                    {
+                        if (x.IsFaulted && _logger.IsError)
+                        {
+                            _logger.Error($"Error during peer storage commit: {x.Exception}");
+                        }
+                        _peerPersistenceTimer.Enabled = true;
+                    });
                 }
                 catch (Exception exception)
                 {
                     if (_logger.IsDebug) _logger.Error("Peer persistence timer failed", exception);
-                }
-                finally
-                {
                     _peerPersistenceTimer.Enabled = true;
                 }
             };
 
             _peerPersistenceTimer.Start();
         }
 
-        private void RunPeerCommit()
+        private async Task RunPeerCommit()
         {
+            await Task.Yield();
```

Comment:
why yield?

Reply:
```
_peerStorage.Commit();
_peerStorage.StartBatch();
```
Will block for a time; hence previous runly in `Task.Run` (though then `.Wait()`ed so blocks anyway 🤷‍♂️). `Task.Yield` requeues it on ThreadPool thread allowing the timer action to complete and execute the next item in timer queue.